testbed/interfaces/node.go

```diff
@@ -0,0 +1,145 @@
+package testbedi
+
+import (
+	"context"
+	"io"
+)
+
+// NewNodeFunc constructs a node implementing the Core interface. It is provided
+// a path to an already created directory `dir`, as well as a map of attributes
+// which can be supplied to shape process execution.
+// Examples of attributes include: which binary to use, docker image, cpu/ram
+// limits, or any other information that may be required to property setup or
+// manage the node.
+type NewNodeFunc func(dir string, attrs map[string]interface{}) (Core, error)
+
+// GetAttrListFunc returns a list of attribute names that can be queried from
+// the node. These attributes may include those can be set from the NewNodeFunc,
+// or additional attributes at may only be available after initialization.
+// Attributes returned should be queriable through the Attribute interface.
+// Examples include: api address, peerid, cpu/ram limits, jitter.
+type GetAttrListFunc func() []string
+
+// GetAttrDescFunc returns the description of the attribute `attr`
+type GetAttrDescFunc func(attr string) (string, error)
+
+type Libp2p interface {
+	// PeerID returns the peer id
+	PeerID() (string, error)
```

Comment:
Any reason not to use actual peer IDs here instead of strings?

Reply:
I believe the reasoning was to make dealing with dependencies easier, e.g. 2 different plugins exists, one uses `go-cid` version foo, another plugin uses `go-cid` version bar. This means iptb will need to be built specifically with the version of `go-cid` the plugin requires, which is annoying. I believe the same holds true for multiformats.